### PR TITLE
Add RSS feed for What's New content

### LINKED ISSUE #1630: Add RSS feed for What's new data
### Is your feature request related to a problem? Please describe

We've had a few requests for RSS for What's New.

### References 

https://github.com/newrelic/docs-website/wiki/Site-Architecture#-docs-as-a-service-json-and-xml-endpoints

### Describe the solution you'd like

* Add an RSS endpoint for What's new data
* Add an RSS UI element to [whats new](https://docs.newrelic.com/whats-new/) for finding the RSS feed
* document this new feed [here](https://github.com/newrelic/docs-website/wiki/Site-Architecture#-docs-as-a-service-json-and-xml-endpoints) 

### Describe why this important to you

Support and solutions folks have been in touch to ask for this. I suspect users would find it helpful as well.

### PR DESCRIPTION
**Pull request by roadlittledawn**
## Description

Adds local plugin to generate RSS for what's new content, a la the way we do it for release notes. Also adds RSS link to [Whats New index page](https://docs.newrelic.com/whats-new) page.

I also validated it via [W3C RSS validator](https://validator.w3.org/feed/)

## Screenshots
Screen Shot 2021-04-29 at 08 29 37
Screen Shot 2021-04-29 at 08 29 49

## Related issues
Closes #1630

### CODE REVIEW

File: plugins/gatsby-plugin-whats-new-rss/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  } = node;
+  const { title, releaseDate } = frontmatter;
+
+  const html = htmlParser.runSync(htmlAst);
+
+  // time is necessary for RSS validity
+  const date = parseISO(releaseDate);
+  const pubDate = `${format(date, 'EE, dd LLL yyyy')} 00:00:00 +0000`;
+  const link = new URL(slug, siteMetadata.siteUrl).href;
+  const id = Buffer.from(title).toString('base64');
```
Comment (zstix) [inline plugins/gatsby-plugin-whats-new-rss/gatsby-node.js:66]: We might want to add the `releaseDate` (in addition to `title`) to ensure that each ID is unique. I could see two posts having the same title (e.g. something like "Updates to Browser Agent").

File: plugins/gatsby-plugin-whats-new-rss/gatsby-node.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  return {
+    guid: id,
+    title,
+    custom_elements: [
+      { link },
+      { pubDate },
+      { 'content:encoded': html },
+      { description: `ReleasedOn: ${pubDate}.` },
```
Comment (zstix) [inline plugins/gatsby-plugin-whats-new-rss/gatsby-node.js:75]: We might be able to use `summary` for this field.

File: plugins/gatsby-plugin-whats-new-rss/gatsby-node.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    ],
+  };
+};
+
+const generateFeed = (publicDir, siteMetadata, reporter, whatsNewNodes) => {
+  const title = `What's new in New Relic`;
```
Comment (zstix) [inline plugins/gatsby-plugin-whats-new-rss/gatsby-node.js:81]: Nitpick, and correct me if I'm wrong @austin-schaefer, I think we're not using a `'` with these: `Whats new`.
```suggestion
  const title = `Whats new in New Relic`;
```

**Review by zstix — COMMENTED** [review]
A couple of minor comments, but this looks great!!

**Comment by austin-schaefer** [comment]
Woop woop! I know sales and support will be excited about this!

File: plugins/gatsby-plugin-whats-new-rss/gatsby-node.js (hunk 1 above)
Comment (roadlittledawn) [reply plugins/gatsby-plugin-whats-new-rss/gatsby-node.js:66]: yeah i wiffle-waffled on what to use. also wasn't sure how many chars is too many for an ID. but `releaseDate` shouldn't add too many more and seems like a good idea to me.

File: plugins/gatsby-plugin-whats-new-rss/gatsby-node.js (hunk 2 above)
Comment (roadlittledawn) [reply plugins/gatsby-plugin-whats-new-rss/gatsby-node.js:75]: good call!

File: plugins/gatsby-plugin-whats-new-rss/gatsby-node.js (hunk 3 above)
Comment (mmfred) [reply plugins/gatsby-plugin-whats-new-rss/gatsby-node.js:81]: What's new should have the apostrophe. Is there a reason that doesn't work in this context?

**Review by zstix — APPROVED** [review]

**Pull request merged**